elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +267,255 @@ private Object fetchRelationship(Environment context) {
+                    Optional.empty(), Optional.empty(), Optional.empty());
+        } catch(BadRequestException e) { /* no id(s) provided, must create new */
+            log.debug(e.toString());
+            return createObject(input, requestScope, Optional.empty(), outputType);
+        }
+
+        if(fetchEntries.isEmpty()) { /* empty set returned, must create new */
+            return createObject(input, requestScope, Optional.ofNullable(idField), outputType);
+        } else { /* must update object */
+            return updateObject(fetchEntries, input);
+        }
+    }
+
+    /**
+     * Creates a new "root-level" resource in the database
+     * @param input individual input data objects
+     * @param outputType graphql-java output type
+     * @param requestScope request scope
+     * @param uuid uuid
+     * @return {@link PersistentResource} object
+     */
+    private Object createObject(Map<String, Object> input, RequestScope requestScope,
+                                Optional<String> uuid, GraphQLType outputType) {
+        GraphQLObjectType objectType = (GraphQLObjectType) ((GraphQLList) outputType).getWrappedType();
+        EntityDictionary dictionary = requestScope.getDictionary();
+        Class<?> entityClass = dictionary.getEntityClass(objectType.getName());
+        Object toCreate = PersistentResource.createObject(null, entityClass, requestScope, uuid);
+        input.entrySet().stream()
+                .filter(entry -> dictionary.isAttribute(entityClass, entry.getKey()))
+                .forEach(entry -> ((PersistentResource) toCreate).updateAttribute(entry.getKey(), entry.getValue()));
+
+        return updateObject(new HashSet<>(Arrays.asList(toCreate)), input);
+    }
+
+    /**
+     * Creates a new "non root-level" resource in the database
+     * @param input individual input data objects
+     * @param requestScope request scope
+     * @param uuid uuid
+     * @param field field name
+     * @param parent parent resource
+     * @return {@link PersistentResource} object
+     */
+    private Object createObject(Map<String, Object> input, RequestScope requestScope,
+                                Optional<String> uuid, Field field, PersistentResource parent) {
+        EntityDictionary dictionary = requestScope.getDictionary();
+        Class<?> entityClass = dictionary.getParameterizedType(parent.getResourceClass(), field.getName());
+        Object toCreate = PersistentResource.createObject(parent, entityClass, requestScope, uuid);
+        input.entrySet().stream()
+                .filter(entry -> dictionary.isAttribute(entityClass, entry.getKey()))
+                .forEach(entry -> ((PersistentResource) toCreate).updateAttribute(entry.getKey(), entry.getValue()));
+
+        //add new relation between parent, updated object
+        parent.addRelation(field.getName(), (PersistentResource) toCreate);
+
+        return updateObject(new HashSet<>(Arrays.asList(toCreate)), input);
```

Comment:
You need to update the relationships next.  To do that, you'll need to extract the value of the data object and recurse by calling upsert again